app/src/main/java/com/example/aboutme/MainActivity.kt

```diff
@@ -0,0 +1,32 @@
+package com.example.aboutme
+
+import android.content.Context
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.view.View
+import android.view.inputmethod.InputMethodManager
+import android.widget.Button
+import android.widget.TextView
+
+class MainActivity : AppCompatActivity() {
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_main)
+
+        findViewById<Button>(R.id.done_button).setOnClickListener{
+            addNickName(it)
+        }
+    }
+
+    private fun addNickName(view: View){
```

Comment:
This function is doing to much things. Let's try to extract some code in another functions